### PR TITLE
Add rewrite parser to include strings and vars

### PR DESCRIPTION
**Pull request by nikklassen**
This works towards #29. It adds strings and env vars but no user vars yet

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    let mut f = true;
+    while let Some(c) = loop_cond(reader) {
+        let mut token = None;
+        for tokenizer in tokenizers.iter() {
+            token = try!(tokenizer(reader));
+            if token.is_some() {
+                break;
+            }
+        }
+        if f {
```
Comment (nikklassen) [inline src/tokenizer.rs]: Remove

File: src/tokenizer.rs [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+fn tokenize_loop<F: Fn(&mut CharReader) -> Option<char>>(
+    reader: &mut CharReader,
+    tokenizers: Vec<fn(&mut CharReader) -> ASTResult>,
+    loop_cond: F)
+-> Result<Vec<AST>, String> {
+
+    let mut tokens = Vec::<AST>::new();
+    let mut word = String::new();
+
+    let mut f = true;
```
Comment (danReynolds) [inline src/tokenizer.rs]: Wasn't this for debugging?

File: src/tokenizer.rs [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+            }
+            contents.push(c);
+        } else {
+            return Err("Unterminated string".to_string());
+        }
+    }
+}
+
+fn quotemark_tok(reader: &mut CharReader) -> ASTResult {
+    if let Some('"') = reader.current {
```
Comment (nikklassen) [inline src/tokenizer.rs]: Invert

File: src/interpolate.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+fn expand_arg(arg: AST, user_env: &UserEnv) -> AST {
+    match arg {
+        AST::Var(v) => {
+            let s = user_env.vars
+                .get(&v)
+                .map(|value| value.clone())
+                .unwrap_or_else(|| {
+                    env::var(&v)
+                    .unwrap_or("".to_string())
```
Comment (nikklassen) [inline src/interpolate.rs]: Use String::from

File: src/lexer.rs [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    fn start() -> OpPositioner {
+        OpPositioner(0)
+    }
+
+    fn update(&self, p: &mut OpPositioner) {
+        p.0 += 1;
+    }
+}
+
+fn args<I>(input: State<I>) -> primitives::ParseResult<AST, I, AST>
```
Comment (nikklassen) [inline src/lexer.rs]: Rename to arg

**Comment by nikklassen** [comment]
closed #29

**Pull request merged**